prov/efa/src/efa_fabric.c

```diff
@@ -479,13 +489,30 @@ static int efa_get_gdr_support(char *device_name)
 		goto out;
 	}
 
-	ret = fi_read_file(get_sysfs_path(), gdr_path, &gdr_support, sizeof(char));
-	if (ret < 0)
+	memset(gdr_support_str, 0, sizeof(gdr_support_str));
+	ret = fi_read_file(get_sysfs_path(), gdr_path,
+			   gdr_support_str, sizeof(gdr_support_str));
+	if (ret < 0) {
+		EFA_WARN(FI_LOG_FABRIC, "Cannot read sysfs file %s!\n", gdr_path);
+		goto out;
+	}
+
+	if (ret == 0) {
+		EFA_WARN(FI_LOG_FABRIC, "Sysfs file %s is empty\n", gdr_path);
+		ret = -FI_EINVAL;
 		goto out;
+	}
 
-	ret = atoi(&gdr_support);
+	ret = 0;
+	*gdr_support = strtol(gdr_support_str, NULL, 10);
+	if (*gdr_support != 0 && *gdr_support != 1) {
+		EFA_WARN(FI_LOG_FABRIC, "Unexpected value of gdr_suport! gdr_support_str: %s gdr_support: %d\n",
+			 gdr_support_str, *gdr_support);
+		ret = -FI_EINVAL;
+	}
```

Comment:
You should have a goto in this clause, to protect against someone adding code later.

Reply:
Thanks! Added in new revision.